protomap/proto.go

```diff
@@ -544,90 +548,148 @@ func ProtoFromPaths(p proto.Message, vals map[*gpb.Path]interface{}, opt ...Unma
 		return np
 	}
 
-	// directCh is a map between the absolute schema path for a particular value, and
-	// the value specified.
-	directCh := map[*gpb.Path]interface{}{}
-	for p, v := range vals {
-		absPath := &gpb.Path{
-			Elem: append(append([]*gpb.PathElem{}, schemaPath(valPrefix).Elem...), p.Elem...),
-		}
+	findChildren := func(vals map[*gpb.Path]any, valPrefix *gpb.Path, protoPrefix *gpb.Path, directOnly, mustBeChildren bool) (map[*gpb.Path]any, error) {
```

Comment:
same here: suggestion to factor this out since it has a full set of parameters already specified.

Reply:
I see it's there in the next PR